chia/consensus/block_record.py

```diff
@@ -41,6 +41,41 @@ def is_transaction_block(self) -> bool:
         return self.timestamp is not None
 
 
+@streamable
+@dataclass(frozen=True)
+class BlockRecordDB(Streamable):
+    """
+    This class contains the fields from `BlockRecord` that get stored in the DB.
+    Unlike `BlockRecord`, this should never extend with more fields, in order to avoid DB corruption.
+    """
+
+    header_hash: bytes32
+    prev_hash: bytes32
+    height: uint32
+    weight: uint128
+    total_iters: uint128
+    signage_point_index: uint8
+    challenge_vdf_output: ClassgroupElement
+    infused_challenge_vdf_output: Optional[ClassgroupElement]
+    reward_infusion_new_challenge: bytes32
+    challenge_block_info_hash: bytes32
+    sub_slot_iters: uint64
+    pool_puzzle_hash: bytes32
+    farmer_puzzle_hash: bytes32
+    required_iters: uint64
+    deficit: uint8
+    overflow: bool
+    prev_transaction_block_height: uint32
+    timestamp: Optional[uint64]
+    prev_transaction_block_hash: Optional[bytes32]
+    fees: Optional[uint64]
+    reward_claims_incorporated: Optional[List[Coin]]
+    finished_challenge_slot_hashes: Optional[List[bytes32]]
+    finished_infused_challenge_slot_hashes: Optional[List[bytes32]]
+    finished_reward_slot_hashes: Optional[List[bytes32]]
+    sub_epoch_summary_included: Optional[SubEpochSummary]
```

Comment:
I think you should factor out this change into a separate PR. the change where you introduce `BlockRecordDB`. I think it's a good, independent, change.

Also, I think this type belongs in `block_store.py` and should be private in there